### PR TITLE
Improve recreate check by looking for .tox-config1

### PR DESCRIPTION
**Pull request by asottile**

### CODE REVIEW

File: src/tox/venv.py [hunk 1]
```diff
@@ -707,6 +707,8 @@ def cleanup_for_venv(venv):
         not venv.path.exists()
         # does exist, but it's empty => OK
         or not dir_items
+        # tox has marked this as an environment it has created in the past
+        or ".tox-config1" in dir_items
         # it exists and we're on windows with Lib and Scripts => OK
```
Comment (gaborbernat) [inline src/tox/venv.py]: Can't we refuse to delete anything without the tox config file? Simplify checks.

Comment (asottile) [reply src/tox/venv.py]: I wanted to continue to support a usecase like:

```bash
virtualenv venv
tox -r --devenv venv
```

Comment (gaborbernat) [reply src/tox/venv.py]: Seems very arcane use case 🤔

Comment (asottile) [reply src/tox/venv.py]: I think it's a pretty realistic one, maybe not directly in succession but consider someone who set up a project which used `make` to manage a virtualenv, but then wanted to switch to use `tox --devenv` to manage that same environment instead.

Comment (gaborbernat) [reply src/tox/venv.py]: Any ideas why the ci fails now?

Comment (gaborbernat) [reply src/tox/venv.py]: https://dev.azure.com/toxdev/tox/_build/results?buildId=2045&view=results take 2

Comment (asottile) [reply src/tox/venv.py]: weird

Comment (rpkilby) [reply src/tox/venv.py]: For `--devenv` can't we also add a `.tox-config1` file to the venv?

Comment (asottile) [reply src/tox/venv.py]: it does, yes -- `--devenv` invokes the normal tox environment provisioning mechanisms

**Comment by asottile** [comment]
I manually triggered this run of the same PR, it didn't update the PR status but it did pass: https://dev.azure.com/toxdev/tox/_build/results?buildId=2048&view=results

File: src/tox/venv.py (hunk 1 above)
Comment (rpkilby) [reply src/tox/venv.py]: Gotcha - missed the part where the env was created prior to devenv. Need ☕️

**Pull request merged**